Lib/nntplib.py

```diff
@@ -1074,27 +1076,19 @@ def __init__(self, host, port=NNTP_SSL_PORT,
             """This works identically to NNTP.__init__, except for the change
             in default port and the `ssl_context` argument for SSL connections.
             """
-            sys.audit("nntplib.connect", self, host, port)
-            self.sock = socket.create_connection((host, port), timeout)
-            file = None
+            self.ssl_context = ssl_context
+            super().__init__(host, port, user, password, readermode,
+                             usenetrc, timeout)
+
+        def _create_socket(self, timeout):
+            self.sock = super()._create_socket(timeout)
```

Comment:
In NNTP, the method only creates a socket, it doesn't assign self.sock. Please replace `self.sock` with `sock`.